lib/workers/global/config/parse/index.ts

```diff
@@ -33,6 +50,18 @@ export async function parseConfigs(
 
   config = await codespaces.setConfig(config);
 
+  let resolvedGlobalExtends: AllConfig | undefined;
+
+  if (is.nonEmptyArray(config?.globalExtends)) {
+    // resolve global presets immediately
+    resolvedGlobalExtends = await resolveGlobalExtends(
```

Comment:
we don't have any global config yet, so things like host rules are not applied while resolving

Reply:
When quickly checking, I don't see any authentication happening in the preset-fetching code, but maybe I'm mistaken.
For us it probably always worked because in our internal network the repo with the presets is not private.

Either way, it looks like this was problem even before the globalExtends bug came up, which we are trying to fix here.

Reply:
If globalExtends previously didn't use auth then we can merge this then

Reply:
Here's why I think that the host rules were not considered for resolving the global presets even before the changes of this PR: https://github.com/renovatebot/renovate/blob/e09764fceceb55cbea88ab8653039e1e7aa8b974/lib/workers/global/index.ts#L139-L154
`globalInitialize(config)` is called only after `resolveGlobalExtends(...)` is called. The `globalInitialize` method contains the [initialization](https://github.com/renovatebot/renovate/blob/6946d5b0ecab7706f2eb804beb82b1a6c98a7510/lib/workers/global/initialize.ts#L85) of the host rules.

Reply:
Confirmed via our _existing_ docs:

![image](https://github.com/user-attachments/assets/8c96d88c-d950-49e2-b46f-4d674ff73667)